flytekit/core/python_function_task.py

```diff
@@ -214,6 +277,14 @@ def __init__(
             task_config=task_config,
             **kwargs,
         )
+        if self.task_resolver is None:
+            if not istestfunction(func=task_function) and isnested(func=task_function):
+                raise ValueError(
+                    "TaskFunction cannot be a nested/inner or local function. "
+                    "It should be accessible at a module level for Flyte to execute it. Test modules with "
+                    "names begining with `test_` are allowed to have nested tasks. If you want to create your own tasks"
+                    "use the TaskResolverMixin"
+                )
```

Comment:
1- what happens if task_resolver is None but the task is not nested?
2- what happens if task_resolver is not None but the task is nested (not in a workflow)?